File: src/web/frontend/robot_apps_flutter/packages/middleware_api_utilities/lib/src/models/drawer.dart
class Drawer {
  final String robotName;
  final int moduleId;
  final int drawerId;
  final int size;

  final Map<String, int> content;
  final List<String> reservedForIds;
  final List<String> reservedForGroups;
  final String moduleProcessStatus;
  final String moduleProcessType;
  final Map<String, int> moduleProcessPayload;
  Drawer({
    required this.robotName,
    required this.moduleId,
    required this.drawerId,
    required this.size,
    required this.content,
    required this.reservedForIds,
    required this.reservedForGroups,
    required this.moduleProcessStatus,
    required this.moduleProcessType,
    required this.moduleProcessPayload,
  });

  // Factory method to create a Drawer instance from a JSON object
  factory Drawer.fromJson(Map<String, dynamic> json) {
    return Drawer(
      robotName: json['robot_name'],
      moduleId: json['module_id'],
      drawerId: json['drawer_id'],
      size: json['size'],
      content: Map<String, int>.from(json['content']),
      reservedForIds: List<String>.from(json['reserved_for_ids']),
      reservedForGroups: List<String>.from(json['reserved_for_groups']),
      moduleProcessStatus: json['module_process_status'],
      moduleProcessType: json['module_process_type'],
      moduleProcessPayload: Map<String, int>.from(json['module_process_payload']),
    );
  }

  // Method to convert a Drawer instance to a JSON object
  Map<String, dynamic> toJson() {
    return {
      'robot_name': robotName,
      'module_id': moduleId,
      'drawer_id': drawerId,
      'size': size,
      'content': content,
      'reserved_for_ids': reservedForIds,
      'reserved_for_groups': reservedForGroups,
      'module_process_status': moduleProcessStatus,
      'module_process_type': moduleProcessType,
      'module_process_payload': moduleProcessPayload,
    };
  }
}

File: src/web/middleware/module_manager/models/drawer.py
from dataclasses import dataclass
from typing import Dict, Any


@dataclass
class Drawer:
    robot_name: str
    module_id: int
    drawer_id: int
    position: int
    size: int
    variant: str
    module_process_status: str
    module_process_type: str
    module_process_payload: Dict[str, int]
    content: Dict[str, int]
    reserved_for_ids: list[str]
    reserved_for_groups: list[str]

    @classmethod
    def from_json(cls, json_data: dict[str, Any]) -> "Drawer":
        return cls(
            robot_name=json_data["robot_name"],
            module_id=json_data["module_id"],
            drawer_id=json_data["drawer_id"],
            size=json_data["size"],
            position=json_data["position"],
            variant=json_data["variant"],
            module_process_status=json_data["module_process_status"],
            module_process_type=json_data["module_process_type"],
            module_process_payload=json_data["module_process_payload"],
            content=json_data["content"],
            reserved_for_ids=json_data["reserved_for_ids"],
            reserved_for_groups=json_data["reserved_for_groups"],
        )

    def to_json(self) -> dict[str, Any]:
        return self.__dict__
